ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -160,10 +168,26 @@ private void putBlob(MessageInfo messageInfo, ByteBuffer messageBuf, long size)
               messageInfo.getSize(), encryptionOrigin, kmsContext, cryptoAgentFactoryClass);
       // If buffer was encrypted, we no longer know its size
       long bufferlen = bufferChanged ? -1 : size;
-      cloudDestination.uploadBlob(blobId, bufferlen, blobMetadata, new ByteBufferInputStream(messageBuf));
+      boolean putDone =
+          cloudDestination.uploadBlob(blobId, bufferlen, blobMetadata, new ByteBufferInputStream(messageBuf));
+      numPutsDone += (putDone ? 1 : 0);
+      numPutsSkipped += (putDone ? 0 : 1);
     } else {
       logger.trace("Blob is skipped: {}", messageInfo);
       vcrMetrics.blobUploadSkippedCount.inc();
+      numPutsSkipped++;
+    }
+  }
+
+  /**
+   * Log the collected put counts on the prescribed interval.
+   */
+  private void logPutCountsPeriodically() {
```

Comment:
Why not use a Guage?  Guage can dispaly in-memory variables. 

Actually, we have all these vars in azureMetrics
```
  public final Counter blobUploadRequestCount;
  public final Counter blobUploadSuccessCount;
  public final Counter blobUploadConflictCount;
```
We can have a Guage version of them.

Reply:
These are logged per partition, whereas the sensors are per host.  Adding sensors for each partition will become unwieldy.  This is strictly for debugging.

Reply:
ok.